homeassistant/components/mobile_app/entity.py

```diff
@@ -31,6 +31,10 @@ def __init__(self, config: dict, device: DeviceEntry, entry: ConfigEntry):
                                     config[ATTR_SENSOR_UNIQUE_ID])
         self._entity_type = config[ATTR_SENSOR_TYPE]
         self.unsub_dispatcher = None
+        device_id = self._registration[ATTR_DEVICE_ID]
+        self.entity_id = "{}.{}_{}_{}".format(self._entity_type, DOMAIN,
+                                              device_id,
+                                              config[ATTR_SENSOR_UNIQUE_ID])
 
```

Comment:
Shouldn't we use `config[ATTR_SENSOR_NAME]` as last part?

Reply:
Nah, the sensor name is the pretty name. I want to use the unique ID that the app provides to allow them to have duplicate sensors. It's up to them to handle overlaps.